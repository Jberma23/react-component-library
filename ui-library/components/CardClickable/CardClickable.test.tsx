import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { CardClickable } from ".";

describe(CardClickable, () => {
  it("renders with defaults", () => {
    const { asFragment } = render(<CardClickable>test card</CardClickable>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("appends additional classnames", () => {
    render(<CardClickable className="my-test-class">test card</CardClickable>);
    const cardClass = screen.getByText("test card")?.getAttribute("class");
    expect(cardClass).toContain("my-test-class");
  });

  it("sets the tag name via the as prop", () => {
    render(<CardClickable as="aside">test card</CardClickable>);
    const card = screen.getByText("test card");
    expect(card).toHaveProperty("tagName", "ASIDE");
  });

  it("passes down additional props", async () => {
    const mockClickHandler = jest.fn();
    render(
      <CardClickable as="button" onClick={mockClickHandler}>
        test card
      </CardClickable>
    );
    const cardEl = screen.getByRole("button");
    await userEvent.click(cardEl);
    expect(mockClickHandler).toHaveBeenCalled();
  });

  it("can be disabled when it's a button", async () => {
    const mockClickHandler = jest.fn();
    render(
      <CardClickable as="button" disabled={true} onClick={mockClickHandler}>
        test card
      </CardClickable>
    );
    const cardEl = screen.getByRole("button");
    await userEvent.click(cardEl);
    expect(mockClickHandler).not.toHaveBeenCalled();
    expect(cardEl).toHaveProperty("disabled", true);
  });

  it("can be disabled when it's not a button", async () => {
    const mockClickHandler = jest.fn();
    render(
      <CardClickable as="div" role="button" disabled>
        test card
      </CardClickable>
    );
    const cardEl = screen.getByRole("button");
    await userEvent.click(cardEl);
    expect(mockClickHandler).not.toHaveBeenCalled();
    const disabledAttr = cardEl?.getAttribute("aria-disabled");
    expect(disabledAttr).toEqual("true");
  });
});
