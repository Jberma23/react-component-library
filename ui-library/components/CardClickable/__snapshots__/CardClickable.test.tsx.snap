// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardClickable renders with defaults 1`] = `
<DocumentFragment>
  <button
    class="block rounded-3 hover:shadow-3 focus-visible:outline focus-visible:outline-2 focus-visible:outline-brand cursor-pointer transition duration-default bg-white shadow-1"
  >
    test card
  </button>
</DocumentFragment>
`;
